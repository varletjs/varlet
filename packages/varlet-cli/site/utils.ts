import { get } from 'lodash-es'
import { StyleProvider, type Theme, withSiteConfigNamespace } from '@varlet/cli/client'

export interface Menu {
  doc: string
  text: Record<string, string>
  type: MenuTypes
  useMobile?: boolean
}

export enum MenuTypes {
  TITLE = 1,
  COMPONENT = 2,
  DOCUMENTATION = 3,
}

export function removeEmpty(object: Record<string, string> = {}) {
  return Object.keys(object).reduce((record: Record<string, string>, key) => {
    const value = object[key]
    value && (record[key] = value)
    return record
  }, {})
}

export function isPhone() {
  return /Android|webOS|iPhone|iPod|BlackBerry|Pad/i.test(navigator.userAgent)
}

export function inIframe() {
  return window.self !== window.top
}

export function setTheme(config: Record<string, any>, name: Theme) {
  const styleVars = withSiteConfigNamespace(get(config, name, {}))

  StyleProvider(styleVars)
}

export function utoa(data: string): string {
  return btoa(unescape(encodeURIComponent(data)))
}
