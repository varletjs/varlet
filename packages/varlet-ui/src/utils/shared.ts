import { TimeData } from '../countdown/props'

export interface CacheInstance<T> {
  cache: T[]

  has(key: T): boolean

  add(key: T): void

  remove(key: T): void

  clear(): void
}

export const toNumber = (val: number | string | boolean | undefined | null): number => {
  if (val == null) return 0

  if (isString(val)) {
    val = parseFloat(val)
    val = Number.isNaN(val) ? 0 : val
    return val
  }

  if (isBool(val)) return Number(val)

  return val
}

export const isHTMLSupportImage = (val: string | undefined | null) => {
  if (val == null) {
    return false
  }

  return val.startsWith('data:image') || /\.(png|jpg|gif|jpeg|svg)$/.test(val)
}

export const isHTMLSupportVideo = (val: string | undefined | null) => {
  if (val == null) {
    return false
  }

  return val.startsWith('data:video') || /\.(mp4|webm|ogg)$/.test(val)
}

export const isString = (val: unknown): val is string => typeof val === 'string'

export const isBool = (val: unknown): val is boolean => typeof val === 'boolean'

export const isNumber = (val: unknown): val is number => typeof val === 'number'

export const isPlainObject = (val: unknown): val is Record<string, any> =>
  Object.prototype.toString.call(val) === '[object Object]'

export const isObject = (val: unknown): val is Record<string, any> => typeof val === 'object' && val !== null

export const isArray = (val: unknown): val is Array<any> => Array.isArray(val)

export const isURL = (val: string | undefined | null) => {
  if (!val) {
    return false
  }

  return /^(http)|(\.*\/)/.test(val)
}

export const isEmpty = (val: unknown) =>
  val === undefined || val === null || val === '' || (Array.isArray(val) && !val.length)

export const removeItem = (arr: Array<unknown>, item: unknown) => {
  if (arr.length) {
    const index: number = arr.indexOf(item)
    if (index > -1) {
      return arr.splice(index, 1)
    }
  }
}

export const throttle = (method: any, mustRunDelay = 200): (() => void) => {
  let timer: number
  let start = 0

  return function loop(this: unknown, ...args) {
    const now = Date.now()
    const elapsed = now - start

    if (!start) {
      start = now
    }

    if (timer) {
      window.clearTimeout(timer)
    }

    if (elapsed >= mustRunDelay) {
      method.apply(this, args)
      start = now
    } else {
      timer = window.setTimeout(() => {
        loop.apply(this, args)
      }, mustRunDelay - elapsed)
    }
  }
}

export const createCache = <T>(max: number): CacheInstance<T> => {
  const cache: T[] = []

  return {
    cache,

    has(key: T) {
      return this.cache.includes(key)
    },

    add(key: T) {
      if (this.has(key)) {
        return
      }

      this.cache.length === max && cache.shift()
      this.cache.push(key)
    },

    remove(key: T) {
      this.has(key) && removeItem(this.cache, key)
    },

    clear() {
      this.cache.length = 0
    },
  }
}

export const linear = (value: number): number => value

export const cubic = (value: number): number => value ** 3

export const easeInOutCubic = (value: number): number =>
  value < 0.5 ? cubic(value * 2) / 2 : 1 - cubic((1 - value) * 2) / 2

export function parseFormat(format: string, time: TimeData): string {
  const scannedTimes = Object.values(time)
  const scannedFormats = ['DD', 'HH', 'mm', 'ss']
  const padValues = [24, 60, 60, 1000]

  scannedFormats.forEach((scannedFormat, index) => {
    if (!format.includes(scannedFormat)) {
      scannedTimes[index + 1] += scannedTimes[index] * padValues[index]
    } else {
      format = format.replace(scannedFormat, String(scannedTimes[index]).padStart(2, '0'))
    }
  })

  if (format.includes('S')) {
    const ms = String(scannedTimes[scannedTimes.length - 1]).padStart(3, '0')

    if (format.includes('SSS')) {
      format = format.replace('SSS', ms)
    } else if (format.includes('SS')) {
      format = format.replace('SS', ms.slice(0, 2))
    } else {
      format = format.replace('S', ms.slice(0, 1))
    }
  }

  return format
}

export const dt = (value: unknown, defaultText: string | undefined) => (value == null ? defaultText : value)

export const inBrowser = () => typeof window !== 'undefined'

export const uniq = (arr: Array<any>) => [...new Set(arr)]

export function kebabCase(str: string): string {
  const ret = str.replace(/([A-Z])/g, ' $1').trim()
  return ret.split(' ').join('-').toLowerCase()
}
