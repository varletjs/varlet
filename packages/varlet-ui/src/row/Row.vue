<template>
  <div
    :class="classes(n(), 'var--box')"
    :style="{
      justifyContent: justify,
      alignItems: align,
      margin: average ? `0 -${average}px` : undefined,
    }"
    @click="onClick"
  >
    <slot />
  </div>
</template>

<script lang="ts">
import { defineComponent, computed, watch } from 'vue'
import { props } from './props'
import { useCols } from './provide'
import { toPxNum } from '../utils/elements'
import type { ComputedRef } from 'vue'
import type { RowProvider } from './provide'
import { createNamespace } from '../utils/components'

const { n, classes } = createNamespace('row')

export default defineComponent({
  name: 'VarRow',
  props,
  setup(props) {
    const { cols, bindCols, length } = useCols()
    const average: ComputedRef<number> = computed(() => {
      const gutter: number = toPxNum(props.gutter)
      return gutter / 2
    })

    const computePadding = () => {
      cols.forEach((col) => {
        col.setPadding({ left: average.value, right: average.value })
      })
    }

    const rowProvider: RowProvider = { computePadding }

    watch(() => length.value, computePadding)
    watch(() => props.gutter, computePadding)

    bindCols(rowProvider)

    return { n, classes, average }
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import './row';
</style>
