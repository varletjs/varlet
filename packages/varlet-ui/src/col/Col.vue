<template>
  <div
    :class="
      classes(
        n(),
        n('$--box'),
        [span >= 0, n(`--span-${span}`)],
        [offset, n(`--offset-${offset}`)],
        ...getSize('xs', xs),
        ...getSize('sm', sm),
        ...getSize('md', md),
        ...getSize('lg', lg),
        ...getSize('xl', xl)
      )
    "
    :style="{
      flexDirection: direction,
      justifyContent: padStartFlex(justify),
      alignItems: padStartFlex(align),
      paddingLeft: toSizeUnit(padding.left),
      paddingRight: toSizeUnit(padding.right),
    }"
    @click="handleClick"
  >
    <slot />
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, computed, watch } from 'vue'
import { isPlainObject, toNumber } from '@varlet/shared'
import { props } from './props'
import { useRow } from './provide'
import { padStartFlex, toSizeUnit } from '../utils/elements'
import { createNamespace, call } from '../utils/components'
import type { Ref, ComputedRef } from 'vue'
import type { ColPadding, ColProvider, ColSizeDescriptor } from './provide'

const { n, classes } = createNamespace('col')

export default defineComponent({
  name: 'VarCol',
  props,
  setup(props) {
    const padding: Ref<ColPadding> = ref({ left: 0, right: 0 })
    const span: ComputedRef<number> = computed(() => toNumber(props.span))
    const offset: ComputedRef<number> = computed(() => toNumber(props.offset))
    const { row, bindRow } = useRow()
    const colProvider: ColProvider = {
      setPadding(pad: ColPadding) {
        padding.value = pad
      },
    }

    const getSize = (mode: string, size: string | number | ColSizeDescriptor | undefined) => {
      const classes: string[] = []

      if (size == null) {
        return classes
      }

      if (isPlainObject(size)) {
        const { offset, span } = size

        Number(span) >= 0 && classes.push(n(`--span-${mode}-${span}`))
        offset && classes.push(n(`--offset-${mode}-${offset}`))
      } else {
        Number(size) >= 0 && classes.push(n(`--span-${mode}-${size}`))
      }

      return classes
    }

    const handleClick = (e: Event) => {
      call(props.onClick, e)
    }

    watch([() => props.span, () => props.offset], () => {
      row?.computePadding()
    })

    call(bindRow, colProvider)

    return {
      n,
      classes,
      padding,
      toNumber,
      toSizeUnit,
      getSize,
      span,
      offset,
      handleClick,
      padStartFlex,
    }
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import './col';
</style>
