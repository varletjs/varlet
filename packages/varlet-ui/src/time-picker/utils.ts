import { hours24, hoursAmpm } from './props'
import type { AmPm, AllowedTime } from './props'
import { toNumber } from '@varlet/shared'

type DisableProps = {
  time: number
  format: string
  ampm: AmPm | undefined
  hour: string
  minute: number
  max: string | undefined
  min: string | undefined
  disableHour: Array<string>
  allowedTime: AllowedTime | undefined
}

export const notConvert = (format: string, ampm: AmPm | undefined): boolean => format === '24hr' || ampm === 'am'

export const convertHour = (format: string, ampm: AmPm | undefined, hour: string) => {
  const index = hoursAmpm.findIndex((hourAmpm) => toNumber(hourAmpm) === toNumber(hour))
  const getHour = notConvert(format, ampm) ? hour : hours24[index]

  return {
    hourStr: getHour,
    hourNum: toNumber(getHour),
  }
}

export const getNumberTime = (time: string) => {
  const [hour, minute, second] = time.split(':')

  return {
    hour: toNumber(hour),
    minute: toNumber(minute),
    second: toNumber(second),
  }
}

export const getIsDisableMinute = (values: Omit<DisableProps, 'minute'>): boolean => {
  const { time, format, ampm, hour, max, min, disableHour } = values
  const { hourStr, hourNum } = convertHour(format, ampm, hour)
  let isBetweenMinMax = false
  let isAllow = false

  if (disableHour.includes(hourStr)) return true

  if (max && !min) {
    const { hour: maxHour, minute: maxMinute } = getNumberTime(max)
    isBetweenMinMax = maxHour === hourNum && time > maxMinute
  }

  if (!max && min) {
    const { hour: minHour, minute: minMinute } = getNumberTime(min)

    isBetweenMinMax = minHour === hourNum && time < minMinute
  }

  if (max && min) {
    const { hour: maxHour, minute: maxMinute } = getNumberTime(max)
    const { hour: minHour, minute: minMinute } = getNumberTime(min)

    isBetweenMinMax = (minHour === hourNum && time < minMinute) || (maxHour === hourNum && time > maxMinute)
  }

  if (values.allowedTime?.minutes) isAllow = values.allowedTime?.minutes(time)

  return isBetweenMinMax || isAllow
}

export const getIsDisableSecond = (values: DisableProps): boolean => {
  const { time, format, ampm, hour, minute, max, min, disableHour } = values
  const { hourStr, hourNum } = convertHour(format, ampm, hour)
  let isBetweenMinMax = false
  let isAllow = false

  if (disableHour.includes(hourStr)) return true

  if (max && !min) {
    const { hour: maxHour, minute: maxMinute, second: maxSecond } = getNumberTime(max)

    isBetweenMinMax = (maxHour === hourNum && maxMinute < minute) || (maxMinute === minute && time > maxSecond)
  }

  if (!max && min) {
    const { hour: minHour, minute: minMinute, second: minSecond } = getNumberTime(min)

    isBetweenMinMax = (minHour === hourNum && minMinute > minute) || (minMinute === minute && time > minSecond)
  }

  if (max && min) {
    const { hour: maxHour, minute: maxMinute, second: maxSecond } = getNumberTime(max)
    const { hour: minHour, minute: minMinute, second: minSecond } = getNumberTime(min)

    isBetweenMinMax =
      (maxHour === hourNum && maxMinute < minute) ||
      (minHour === hourNum && minMinute > minute) ||
      (maxHour === hourNum && maxMinute === minute && time > maxSecond) ||
      (minHour === hourNum && minMinute === minute && time < minSecond)
  }

  if (values.allowedTime?.seconds) isAllow = values.allowedTime?.seconds(time)

  return isBetweenMinMax || isAllow
}
