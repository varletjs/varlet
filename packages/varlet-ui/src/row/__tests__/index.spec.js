import Row from '..'
import VarRow from '../Row'
import Col from '../../col'
import VarCol from '../../col/Col'
import { mount } from '@vue/test-utils'
import { createApp, h } from 'vue'
import { delay, mockConsole } from '../../utils/jest'

test('test row and col use', () => {
  const app = createApp({}).use(Row).use(Col)
  expect(app.component(Row.name)).toBeTruthy()
  expect(app.component(Col.name)).toBeTruthy()
})

describe('test row and col component props', () => {
  test('test row and col gutter', () => {
    const wrapper = mount(VarRow, {
      props: {
        gutter: 20,
      },
      slots: {
        default: () => [12, 12].map((span) => h(VarCol, { span })),
      },
    })

    expect(wrapper.find('.var-row').attributes('style')).toContain('margin: 0px -10px;')
    wrapper.unmount()
  })

  test('test row and col gutter computed', async () => {
    const wrapper = mount(VarRow, {
      props: {
        gutter: 20,
      },
      slots: {
        default: () => [12, 12].map((span) => h(VarCol, { span })),
      },
    })

    await delay(0)
    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.setProps({ gutter: 40 })
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })

  test('test row and col justify', () => {
    ;['flex-start', 'flex-end', 'center', 'space-between', 'space-around'].forEach((justify) => {
      const wrapper = mount(VarRow, {
        props: {
          justify,
        },
        slots: {
          default: () => [12, 12].map((span) => h(VarCol, { span })),
        },
      })

      expect(wrapper.find('.var-row').attributes('style')).toContain('justify-content: ' + justify)
      wrapper.unmount()
    })
  })

  test('test row and col align', () => {
    ;['flex-start', 'flex-end', 'center'].forEach((align) => {
      const wrapper = mount(VarRow, {
        props: {
          align,
        },
        slots: {
          default: () => [12, 12].map((span) => h(VarCol, { span })),
        },
      })

      expect(wrapper.find('.var-row').attributes('style')).toContain('align-items: ' + align)
      wrapper.unmount()
    })
  })

  test('test row and col span', () => {
    const wrapper = mount(VarRow, {
      slots: {
        default: () => [6, 6, 6, 6].map((span) => h(VarCol, { span })),
      },
    })

    expect(wrapper.findAll('.var-col--span-6').length).toBe(4)
    wrapper.unmount()
  })

  test('test row and col span 0', () => {
    const wrapper = mount(VarRow, {
      slots: {
        default: () => [0, 12, 12].map((span) => h(VarCol, { span })),
      },
    })

    expect(wrapper.findAll('.var-col--span-0').length).toBe(1)
    wrapper.unmount()
  })

  test('test row and col offset', () => {
    const wrapper = mount(VarRow, {
      slots: {
        default: () => [6, 6].map((span) => h(VarCol, { span, offset: 6 })),
      },
    })

    expect(wrapper.findAll('.var-col--offset-6').length).toBe(2)
    wrapper.unmount()
  })

  test('test row and col responsive', () => {
    const wrapper = mount(VarRow, {
      slots: {
        default: () =>
          h(VarCol, {
            xs: { span: 12 },
            sm: { span: 6 },
            md: { span: 4 },
            lg: { span: 3 },
            xl: { span: 2 },
          }),
      },
    })

    const classNames = wrapper.find('.var-col').classes()
    expect(classNames).toContain('var-col--span-xs-12')
    expect(classNames).toContain('var-col--span-sm-6')
    expect(classNames).toContain('var-col--span-md-4')
    expect(classNames).toContain('var-col--span-lg-3')
    expect(classNames).toContain('var-col--span-xl-2')
    wrapper.unmount()
  })

  test('test col direction', () => {
    const wrapper = mount(VarRow, {
      slots: {
        default: () => [12, 12].map((span) => h(VarCol, { span, direction: 'column' })),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })
})

test('test row and col responsive 0', () => {
  const wrapper = mount(VarRow, {
    slots: {
      default: () => [h(VarCol, { xs: 0, sm: 0, md: 0, lg: 0, xl: 0 })],
    },
  })

  expect(wrapper.find('.var-col--span-xs-0')).toBeTruthy()
  expect(wrapper.find('.var-col--span-sm-0')).toBeTruthy()
  expect(wrapper.find('.var-col--span-md-0')).toBeTruthy()
  expect(wrapper.find('.var-col--span-lg-0')).toBeTruthy()
  expect(wrapper.find('.var-col--span-xl-0')).toBeTruthy()
  wrapper.unmount()
})

test('test col is not used in row', () => {
  const { mockRestore } = mockConsole('warn', (text) => {
    expect(text).toBe('col must in row')
  })

  mount(VarCol)

  mockRestore()
})
