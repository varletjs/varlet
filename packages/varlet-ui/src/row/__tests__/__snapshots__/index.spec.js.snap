// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test col in responsive 1`] = `
"<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-24 var-col--span-sm-12 var-col--span-md-8 var-col--span-lg-6 var-col--span-xl-4\\" style=\\"padding-left: 0px; padding-right: 0px;\\">1</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-24 var-col--span-sm-12 var-col--span-md-8 var-col--span-lg-6 var-col--span-xl-4\\" style=\\"padding-left: 0px; padding-right: 0px;\\">2</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-24 var-col--span-sm-12 var-col--span-md-8 var-col--span-lg-6 var-col--span-xl-4\\" style=\\"padding-left: 0px; padding-right: 0px;\\">3</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-24 var-col--span-sm-12 var-col--span-md-8 var-col--span-lg-6 var-col--span-xl-4\\" style=\\"padding-left: 0px; padding-right: 0px;\\">4</div>
</div>"
`;

exports[`test col in responsive 2`] = `
"<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--span-sm-24 var-col--span-md-12 var-col--span-lg-12 var-col--span-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">1</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--span-sm-24 var-col--span-md-12 var-col--span-lg-12 var-col--span-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">2</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--span-sm-24 var-col--span-md-12 var-col--span-lg-12 var-col--span-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">3</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--span-sm-24 var-col--span-md-12 var-col--span-lg-12 var-col--span-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">4</div>
</div>"
`;

exports[`test col in responsive 3`] = `
"<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--offset-xs-12 var-col--span-sm-12 var-col--offset-sm-12 var-col--span-md-12 var-col--offset-md-12 var-col--span-lg-12 var-col--offset-lg-12 var-col--span-xl-12 var-col--offset-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">1</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--offset-xs-12 var-col--span-sm-12 var-col--offset-sm-12 var-col--span-md-12 var-col--offset-md-12 var-col--span-lg-12 var-col--offset-lg-12 var-col--span-xl-12 var-col--offset-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">2</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--offset-xs-12 var-col--span-sm-12 var-col--offset-sm-12 var-col--span-md-12 var-col--offset-md-12 var-col--span-lg-12 var-col--offset-lg-12 var-col--span-xl-12 var-col--offset-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">3</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--offset-xs-12 var-col--span-sm-12 var-col--offset-sm-12 var-col--span-md-12 var-col--offset-md-12 var-col--span-lg-12 var-col--offset-lg-12 var-col--span-xl-12 var-col--offset-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">4</div>
</div>"
`;

exports[`test col in responsive 4`] = `
"<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-24 var-col--span-sm-12 var-col--offset-sm-12 var-col--span-md-12 var-col--offset-md-12 var-col--span-lg-12 var-col--span-xl-12 var-col--offset-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">1</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-24 var-col--span-sm-12 var-col--offset-sm-12 var-col--span-md-12 var-col--offset-md-12 var-col--span-lg-12 var-col--span-xl-12 var-col--offset-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">2</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-24 var-col--span-sm-12 var-col--offset-sm-12 var-col--span-md-12 var-col--offset-md-12 var-col--span-lg-12 var-col--span-xl-12 var-col--offset-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">3</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-24 var-col--span-sm-12 var-col--offset-sm-12 var-col--span-md-12 var-col--offset-md-12 var-col--span-lg-12 var-col--span-xl-12 var-col--offset-xl-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">4</div>
</div>"
`;

exports[`test col in row 1`] = `
"<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-8 var-col--offset-4\\" style=\\"padding-left: 0px; padding-right: 0px;\\">1</div>
  <div class=\\"var-col var--box var-col--span-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">2</div>
</div>"
`;

exports[`test col in row 2`] = `
"<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">1</div>
  <div class=\\"var-col var--box var-col--span-12\\" style=\\"padding-left: 0px; padding-right: 0px;\\">2</div>
</div>"
`;

exports[`test row Responsive example 1`] = `
"<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start; margin: 0px -5px;\\">
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-8 var-col--span-sm-6 var-col--span-md-4 var-col--span-lg-3 var-col--span-xl-1 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">1</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-4 var-col--span-sm-6 var-col--span-md-8 var-col--span-lg-9 var-col--span-xl-11 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">2</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-4 var-col--span-sm-6 var-col--span-md-8 var-col--span-lg-9 var-col--span-xl-11 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">3</div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-8 var-col--span-sm-6 var-col--span-md-4 var-col--span-lg-3 var-col--span-xl-1 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">4</div>
</div>"
`;

exports[`test row ResponsiveObject example 1`] = `
"<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start; margin: 0px -5px;\\">
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--offset-xs-6 var-col--span-sm-6 var-col--span-md-4 var-col--offset-md-4 var-col--span-lg-3 var-col--offset-lg-3 var-col--span-xl-2 var-col--offset-xl-2 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\"> 1 </div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--offset-xs-6 var-col--span-sm-6 var-col--span-md-4 var-col--offset-md-4 var-col--span-lg-3 var-col--offset-lg-3 var-col--span-xl-2 var-col--offset-xl-2 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\"> 2 </div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--offset-xs-6 var-col--span-sm-6 var-col--span-md-4 var-col--offset-md-4 var-col--span-lg-3 var-col--offset-lg-3 var-col--span-xl-2 var-col--offset-xl-2 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\"> 3 </div>
  <div class=\\"var-col var--box var-col--span-24 var-col--span-xs-12 var-col--offset-xs-6 var-col--span-sm-6 var-col--span-md-4 var-col--offset-md-4 var-col--span-lg-3 var-col--offset-lg-3 var-col--span-xl-2 var-col--offset-xl-2 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\"> 4 </div>
</div>"
`;

exports[`test row example 1`] = `
"<div class=\\"app-type\\">栅格系统</div>
<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
</div>
<div class=\\"app-type\\">偏移</div>
<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-16 var-col--offset-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">offset: 8 span: 16</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
</div>
<div class=\\"app-type\\">对齐</div>
<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
</div>
<div class=\\"var-row var--box\\" style=\\"justify-content: center; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
</div>
<div class=\\"var-row var--box\\" style=\\"justify-content: flex-end; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
</div>
<div class=\\"var-row var--box\\" style=\\"justify-content: space-around; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
</div>
<div class=\\"var-row var--box\\" style=\\"justify-content: space-between; align-items: flex-start;\\">
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
</div>
<div class=\\"app-type\\">列间距</div>
<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start; margin: 0px -5px;\\">
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
  <div class=\\"var-col var--box var-col--span-8 col\\" style=\\"padding-left: 0px; padding-right: 0px;\\">span: 8</div>
</div>"
`;

exports[`test row flex 1`] = `"<div class=\\"var-row var--box\\" style=\\"justify-content: flex-start; align-items: flex-start;\\"></div>"`;

exports[`test row flex 2`] = `"<div class=\\"var-row var--box\\" style=\\"justify-content: flex-end; align-items: flex-start;\\"></div>"`;

exports[`test row flex 3`] = `"<div class=\\"var-row var--box\\" style=\\"justify-content: center; align-items: flex-start;\\"></div>"`;

exports[`test row flex 4`] = `"<div class=\\"var-row var--box\\" style=\\"justify-content: space-between; align-items: flex-start;\\"></div>"`;

exports[`test row flex 5`] = `"<div class=\\"var-row var--box\\" style=\\"justify-content: space-around; align-items: flex-start;\\"></div>"`;

exports[`test row flex 6`] = `"<div class=\\"var-row var--box\\" style=\\"justify-content: space-around; align-items: flex-start;\\"></div>"`;

exports[`test row flex 7`] = `"<div class=\\"var-row var--box\\" style=\\"justify-content: space-around; align-items: center;\\"></div>"`;

exports[`test row flex 8`] = `"<div class=\\"var-row var--box\\" style=\\"justify-content: space-around; align-items: flex-end;\\"></div>"`;
